### DISCUSSION TITLE
pdfgen on Microcontroller

### DISCUSSION THREAD

**Discussion by devtranszio** [AndreRenaud/PDFGen#152] [Q&A]
Hi, we like to create .pdf file in USB stick through 32bit microcontroller. Is this library will support for it? like in Embedded Systems.

**Comment by AndreRenaud** [comment]
There's no reason it wouldn't work, but it does depend on some libc features that you might not have on an embedded system - specifically dynamic memory (`malloc`, `free` etc...) and file operations (`fopen`, `fwrite` etc...). Assuming you've got these, or similar work-alikes, it should be possible to get it running.

If you're making a big PDF, especially if it involves images, be aware that it might require quite a bit of memory. Are you using a standard RTOS? What is the microcontroller?

**Comment by devtranszio** [comment]
Hi Andre Renaud, thanks for the response. We are using Microchip PIC32MX595
series and it has 512Kb of Flash and 96kb of RAM also we did not try with
RTOS. May be, we will try with FreeRTOS.
